src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,8 +1,62 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+
 public class WorkWithFile {
+    private static final String SUPPLY = "supply";
+    private static final String BUY = "buy";
+    private static final String RESULT = "result";
+    private static final String[] values = {SUPPLY, BUY, RESULT};
 
     public void getStatistic(String fromFileName, String toFileName) {
+        List<String> dataFromOutput = getLinesFromFile(fromFileName);
+        HashMap<String, Integer> readyData = getFormattedInformation(dataFromOutput);
+        putInformationInNewFile(readyData, toFileName);
+    }
+
+    private List<String> getLinesFromFile(String fileName) {
+        List<String> listOfInformation = new ArrayList<>();
+        try (BufferedReader reader = new BufferedReader(new FileReader(fileName))) {
+            while (reader.ready()) {
+                listOfInformation.add(reader.readLine());
+            }
+        } catch (FileNotFoundException e) {
+            throw new RuntimeException("file not found", e);
+        } catch (IOException e) {
+            throw new RuntimeException("problem with read file. Try again", e);
+        }
+        return listOfInformation;
+    }
+
+    private HashMap<String, Integer> getFormattedInformation(List<String> data) {
+        HashMap<String, Integer> result = new HashMap<>();
+        for (String line : data) {
+            String[] tempData = line.split(",");
+            result.merge(tempData[0], Integer.parseInt(tempData[1]), Integer::sum);
```

Comment:
that's cool that you know this method and how to use it, but I guess that just count two variables and only then put them in the map once(instead of checking every time if such keys exist) would be much easier

Reply:
Если, скажем появиться пункт "издержки", "уценка", мне прийдется каждый раз добавлять новые сущности для этого пункта.
В этом участке кода - полная абстракция, независимо от того - что мы принимаем.
Насколько я понял, от меня ожидается что-то типо:
result.put(tempData[0], Integer.parseInt(tempData[1]) + result.get(tempData[0]))
Я прав?

Reply:
discussed on QnA, decided to leave as it is